### PR TITLE
[Stats & UserGrowthGraph screens]: Implement pull-to-refresh

### PR DESCRIPTION
**Pull request by ice-atlas**
- implement pull-to-refresh on Stats screen;
- implement pull-to-refresh on UserGrowthGraph screen;
- use new pull-to-refresh on Home screen.

### CODE REVIEW

File: src/screens/News/index.tsx [hunk 1]
```diff
@@ -134,7 +134,8 @@ export const News = () => {
         ]}>
         <RefreshIceIcon
           theme={'light-content'}
-          refreshing={refreshing}
+          // Loading state not implemented
+          refreshing={false}
```
Comment (ice-orion) [inline src/screens/News/index.tsx:153]: But we have that `refreshing` bool that is actually any news loading, don't we? So can we add a loading state on top of that bool (e.g. like in `src/hooks/useRefresh.ts`)?

File: src/screens/HomeFlow/Stats/index.tsx [hunk 2]
```diff
@@ -1,23 +1,76 @@
...
+
   const tabbarOffset = useBottomTabBarOffsetStyle();
 
+  const translateY = useSharedValue(0);
+
+  const scrollHandler = useAnimatedScrollHandler(({contentOffset: {y}}) => {
+    translateY.value = y;
+  });
+
+  const onRefresh = useCallback(() => {
```
Comment (ice-orion) [inline src/screens/HomeFlow/Stats/index.tsx]: Can we please encapsulate this refreshing code in a hook (e.g. `useRefresh`)?

File: src/screens/News/index.tsx (hunk 1 above)
Comment (ice-atlas) [reply src/screens/News/index.tsx:153]: Yes, we can and we have value, but UI does not have loading state (in this case). It needs to be implemented separately

File: src/screens/HomeFlow/Stats/index.tsx (hunk 2 above)
Comment (ice-atlas) [reply src/screens/HomeFlow/Stats/index.tsx]: Done. But can not use `useRefresh` because it wont work with actions where `.create(?)` with `params`

**Review by ice-orion — DISMISSED** [review]

**Review by ice-hector — DISMISSED** [review]

File: src/screens/HomeFlow/UserGrowthGraph/index.tsx [hunk 3]
```diff
@@ -106,52 +128,58 @@ export const UserGrowthGraph = memo(() => {
...
+        <Animated.FlatList
+          showsVerticalScrollIndicator={false}
+          initialNumToRender={14}
+          contentContainerStyle={[
+            styles.contentContainer,
+            tabbarOffset.current,
+          ]}
+          data={data}
+          removeClippedSubviews={false}
+          onLayout={onLayout}
```
Comment (ice-orion) [inline src/screens/HomeFlow/UserGrowthGraph/index.tsx]: Optional: `onLayout` is an inline function but all the others for this component are not. May be for code consistency move `onLayout` function body here? That also saves you from the need to declare types explicitly and hence reduce the code length.
```
onLayout={e => setWidth(e.nativeEvent.layout.width)}
```

Comment (ice-atlas) [reply src/screens/HomeFlow/UserGrowthGraph/index.tsx]: Ok, here make sense

**Review by ice-aiolos — APPROVED** [review]

**Review by ice-hades — APPROVED** [review]

**Review by ice-orion — APPROVED** [review]

**Pull request merged**